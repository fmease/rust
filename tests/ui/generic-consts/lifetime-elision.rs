#![feature(generic_consts)]
#![allow(incomplete_features)]

// Check that we deny lifetime elision inside where-clauses of consts.
// FIXME(generic_consts): Are there other places we should check?

const K<T>: () = ()
where
    &T: Copy; //~ ERROR `&` without an explicit lifetime name cannot be used here

fn main() {}
