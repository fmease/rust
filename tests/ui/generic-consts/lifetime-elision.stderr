error[E0637]: `&` without an explicit lifetime name cannot be used here
  --> $DIR/lifetime-elision.rs:9:5
   |
LL |     &T: Copy;
   |     ^ explicit lifetime name needed here
   |
help: consider introducing a higher-ranked lifetime here
   |
LL |     for<'a> &'a T: Copy;
   |     +++++++  ++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0637`.
